test/functional/cmap/commands.test.ts

```diff
@@ -0,0 +1,122 @@
+import { spy } from 'sinon';
+import { expect } from 'chai';
+import * as BSON from 'bson';
+
+const deserializeSpy = spy(BSON, 'deserialize');
+
+const EXPECTED_VALIDATION_DISABLED_ARGUMENT = {
+  utf8: false
+};
+
+const EXPECTED_VALIDATION_ENABLED_ARGUMENT = {
+  utf8: {
+    writeErrors: false
+  }
+};
+
+describe('class BinMsg', () => {
+  beforeEach(() => {
+    deserializeSpy.resetHistory();
+  });
+
+  describe('enableUtf8Validation option set to false', () => {
+    let client;
+    const option = { enableUtf8Validation: false };
+
+    for (const passOptionTo of ['client', 'db', 'collection', 'operation']) {
+      it(`should disable validation with option passed to ${passOptionTo}`, async function () {
+        try {
+          client = this.configuration.newClient(passOptionTo === 'client' ? option : undefined);
+          await client.connect();
+
+          const db = client.db(
+            'bson_utf8Validation_db',
+            passOptionTo === 'db' ? option : undefined
+          );
+          const collection = db.collection(
+            'bson_utf8Validation_coll',
+            passOptionTo === 'collection' ? option : undefined
+          );
+
+          await collection.insertOne(
+            { name: 'John Doe' },
+            passOptionTo === 'operation' ? option : {}
+          );
+
+          expect(deserializeSpy.called).to.be.true;
+          const validationArgument = deserializeSpy.lastCall.lastArg.validation;
+          expect(validationArgument).to.deep.equal(EXPECTED_VALIDATION_DISABLED_ARGUMENT);
+        } finally {
+          await client.close();
+        }
```

Comment:
Nothing to change here, just calling out that normally we would like to use  an `afterEach` hook to ensure the client closes when each test finishes, unfortunately our session leak checker is currently implemented as an afterEach hook that runs before the one you would define here. So your hook wouldn't get the chance to clean up the sessions/client before the leak checker would throw.